### PR TITLE
Get user

### LINKED ISSUE #3: Add trip information to GET users/:id
should be an array of trip objects for all trips belonging to that user

### PR DESCRIPTION
**Pull request by GregoryJFischer**
### What'd ya do to the code?
Closes #3 
- finished up initial endpoints for users and trips
- refactored urls and views to be handwritten as opposed to autogenerated
- when a user creates a trip they are automatically added to the trip's set of users
- added happy path request testing for all endpoints

### Select One Thing
- [x] This broke nothing
- [ ] This broke something
- [ ] This broke everything

### Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor(DRY-ing up code, removing comments etc.)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:
- [x] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [ ] There are comments where clarification/ organization is needed
- [x] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

### What's next?
- Need to figure out how available dates is going to work (and then implement it too)
- Error handling
- Sad path / edge case testing
- model testing
- serializer testing

### CODE REVIEW

File: api/tests.py [hunk 1]
```diff
@@ -1,13 +1,154 @@
+from urllib import response
 from django.db import IntegrityError
 from django.test import TestCase
 from api.models import User, Trip
 # TestCase is a subclass of unittest, the default Python testing framework
 
 class URLTests(TestCase):
```
Comment (samueldevine) [inline api/tests.py:7]: great work testing these requests! i think that it might not be best practice to have everything in one file - maybe we can refactor our test suite later. but everything here is readable and makes sense to me, nice job

File: api/views.py [hunk 2]
```diff
@@ -1,27 +1,90 @@
...
 from rest_framework.response import Response
+from rest_framework.decorators import api_view
 
 from .serializers import UserSerializer, UserTripSerializer, TripSerializer
 from .models import User, Trip
 
-class UserViewSet(viewsets.ModelViewSet):
-    queryset = User.objects.all()
-    serializer_class = UserSerializer
+@api_view(['GET', 'POST'])
```
Comment (samueldevine) [inline api/views.py:8]: can you describe what this line is doing either here or on zoom tomorrow? i have _some_ understanding from googling but i'm still a little confused

**Comment by samueldevine** [comment]
looks amazing overall! would love to spend some more time with you going over the `views` just to solidify my own understanding. holding off on merging so that we can hopefully have khoa take a look as well.

File: api/tests.py (hunk 1 above)
Comment (GregoryJFischer) [reply api/tests.py:7]: definitely, I figured for now I would just get the tests written, and then it would be pretty easy to just move the tests to appropriate files

File: api/views.py (hunk 2 above)
Comment (GregoryJFischer) [reply api/views.py:8]: essentially that specifies what type of request is allowed. so for the url this view is associated with, the only allowed methods are GET and POST. The other types of requests return a HTTP error (I believe that it's 405)
